Type exposed using `InternalsVisibleTo` is not available when using `5.0.100-preview.6.20314.3` SDK

PowerShell has a dependency on `Microsoft.Management.Infrastructure.dll` which has an internal type `CimMofDeserializer`. It has `System.Management.Automation` in the list of `InternalsVisibleTo`. 

When trying to build `System.Management.Automation` we see the following errors:

```
D:\PSGit\PowerShell\src\System.Management.Automation\DscSupport\CimDSCParser.cs(323,17): error CS0122: 'CimMofDeserializer' is inaccessible due to its protection level [D:\PSGit\PowerShell\src\System.Management.Automation\System.Management.Automation.csproj]
D:\PSGit\PowerShell\src\System.Management.Automation\DscSupport\CimDSCParser.cs(324,17): error CS0122: 'CimMofDeserializer' is inaccessible due to its protection level [D:\PSGit\PowerShell\src\System.Management.Automation\System.Management.Automation.csproj]
D:\PSGit\PowerShell\src\System.Management.Automation\DscSupport\CimDSCParser.cs(327,31): error CS0122: 'CimMofDeserializer' is inaccessible due to its protection level [D:\PSGit\PowerShell\src\System.Management.Automation\System.Management.Automation.csproj]
D:\PSGit\PowerShell\src\System.Management.Automation\DscSupport\CimDSCParser.cs(335,31): error CS0122: 'CimMofDeserializer' is inaccessible due to its protection level [D:\PSGit\PowerShell\src\System.Management.Automation\System.Management.Automation.csproj]
D:\PSGit\PowerShell\src\System.Management.Automation\DscSupport\CimDSCParser.cs(335,129): error CS0122: 'MofDeserializerSchemaValidationOption' is inaccessible due to its protection level [D:\PSGit\PowerShell\src\System.Management.Automation\System.Management.Automation.csproj]
```

This only happens when we upgrade the SDK version from `5.0.100-preview.5.20279.10` to `5.0.100-preview.6.20314.3`

Repro steps:

1. Clone repository https://github.com/PowerShell/PowerShell.git
1. Update the global.json file to have `5.0.100-preview.6.20314.3`
1. In PowerShell (on Windows), `Import-Module build.psm1`
1. `Start-PSBootstrap` to install the Preview 6 SDK
1. Start-PSBuild -Clean

--- a/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets
+++ b/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets
@@ -891,7 +891,7 @@ Copyright (c) .NET Foundation. All rights reserved.
   ============================================================
   -->
   <Target Name="_BlockWinMDsOnUnsupportedTFMs"
-          BeforeTargets="CoreCompile"
+          AfterTargets="PreBuildEvent"
           DependsOnTargets="ResolveReferences"
           Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), '5.0'))">
     <NETSdkError Condition="'%(ReferencePath.Extension)' == '.winmd'"

